### PR TITLE
Flag interfaces

### PR DESCRIPTION
**Pull request by hautajoki**
this pr adds support for flag interfaces not associated with any component data.
```ts
`// A component that stores data
interface Position {
    x: number;
    y: number;
}

// A flag (contains no data)
interface Alive extends Flag {}

const ball = spawn();

// Components with data must be assigned using set()
set<Position>(ball, { x: 10, y: 20 });

// @ts-expect-error: Cannot use add() with components that contain data
add<Position>(ball);

// @ts-expect-error: Cannot use set() with flag components
set<Alive>(ball, true);

// Tag components can only be added or removed
add<Alive>(ball);

// Flags can also be used  with spawn/insert functions
const cube = spawn<[Alive]>()

const torus = spawn()
insert<[Position, Alive]>([{ x: 10, y: 20 }])
```

### CODE REVIEW

**Comment by Ukendio** [comment]
I dont like that it is called a Flag and not a Tag. It doesnt really represent the same things. You can use Tags as flags for certain things, but can also be representative of symbolic state.

File: src/registry.ts [hunk 1]
```diff
@@ -462,7 +489,7 @@ export function parent(entity: Entity): Entity | undefined {
  * @returns The pair ID.
  * @metadata macro
  */
-export function pair<P>(object: Entity, predicate?: ComponentKey<P>): Pair<P, unknown> {
+export function pair<P>(object: Entity, predicate?: ComponentKey<P>): Pair<OmitFlag<P>, unknown> {
```
Comment (Ukendio) [inline src/registry.ts:492]: Just saw thiis but the first element is the predicate not the object.

File: src/registry.ts [hunk 2]
```diff
@@ -3,8 +3,12 @@ import * as ecs from "@rbxts/jecs";
 
 import { createSignal, type Signal } from "./signal";
 
-export interface Wildcard {}
-export interface ChildOf {}
+export interface Flag {
```
Comment (Ukendio) [inline src/registry.ts:6]: Change the name of this interface and every subsequent dependents on this interface to use "Tag"

**Review by Ukendio — CHANGES_REQUESTED** [review]

File: src/registry.ts (hunk 2 above)
Comment (hautajoki) [reply src/registry.ts:6]: Tag is already used for Entity<.undefined>. Should I just get rid of that then?

File: src/registry.ts (hunk 1 above)
Comment (hautajoki) [reply src/registry.ts:492]: The predicate here is a flamework generated key. `pair<SomeComponent>(object)`

File: src/registry.ts (hunk 2 above)
Comment (Ukendio) [reply src/registry.ts:6]: > Should I just get rid of that then?

Yeah, that sounds reasonable.

**Pull request closed without merging**